File: pantallote.py
import tkinter as tk
import random
from PIL import Image, ImageTk
import time

# Crear la ventana principal
root = tk.Tk()
root.title("Pantalla de Espera")
root.geometry("900x900")  

# Fondo verde
root.configure(bg="green")

# Cargar las imágenes
image1 = Image.open("knekro1.png")  
image2 = Image.open("knekro2.png")  

# Convertir las imágenes a formato adecuado para Tkinter
image1_tk = ImageTk.PhotoImage(image1)
image2_tk = ImageTk.PhotoImage(image2)

# Crear un lienzo para dibujar las imágenes
canvas = tk.Canvas(root, width=900, height=900, bg="green", bd=0, highlightthickness=0)
canvas.pack()

# Función para actualizar la posición de las imágenes
 
X = True
def update_position():
    # Borrar las imágenes anteriores
    canvas.delete("all")
    global X
    # Elegir aleatoriamente una imagen y una posición
    #selected_image = image1_tk if random.choice([True, False]) else image2_tk
    if X:
    	selected_image = image1_tk
    	X = False
    else:
    	selected_image = image2_tk
    	X = True	
    x_pos = random.randint(0, 550)  # Coordenada X aleatoria
    y_pos = random.randint(0, 700)  # Coordenada Y aleatoria
    
    # Dibujar la imagen en la nueva posición
    canvas.create_image(x_pos, y_pos, image=selected_image)
    
    # Llamar nuevamente a esta función después de un pequeño retraso
    root.after(200, update_position)  # 200 ms de retraso para crear la animación

# Iniciar la animación
update_position()

# Ejecutar la ventana
root.mainloop()

File: pantallote.cpp
#include <SFML/Graphics.hpp>
#include <SFML/Window.hpp>
#include <cstdlib>
#include <ctime>

int main() {
    // Crear la ventana
    sf::RenderWindow window(sf::VideoMode(900, 900), "Pantalla de Espera");

    // Establecer el fondo verde
    window.clear(sf::Color::Green);

    // Cargar las imágenes
    sf::Texture texture1, texture2;
    if (!texture1.loadFromFile("knekro1.png") || !texture2.loadFromFile("knekro2.png")) {
        return -1; // Error al cargar las imágenes
    }

    // Convertir las texturas en sprites
    sf::Sprite sprite1(texture1);
    sf::Sprite sprite2(texture2);

    // Control de la imagen actual
    bool X = true;

    // Semilla para aleatoriedad
    std::srand(static_cast<unsigned int>(std::time(0)));

    // Temporizador para la animación
    sf::Clock clock;

    // Bucle principal
    while (window.isOpen()) {
        sf::Event event;
        while (window.pollEvent(event)) {
            if (event.type == sf::Event::Closed) {
                window.close();
            }
        }

        // Borrar la ventana con el fondo verde
        window.clear(sf::Color::Green);

        // Elegir aleatoriamente la imagen
        sf::Sprite* selected_sprite = nullptr;
        if (X) {
            selected_sprite = &sprite1;
            X = false;
        } else {
            selected_sprite = &sprite2;
            X = true;
        }

        // Asignar una posición aleatoria para la imagen
        float x_pos = static_cast<float>(std::rand() % 551);  // Coordenada X aleatoria
        float y_pos = static_cast<float>(std::rand() % 701);  // Coordenada Y aleatoria

        // Establecer la posición de la imagen seleccionada
        selected_sprite->setPosition(x_pos, y_pos);

        // Dibujar la imagen
        window.draw(*selected_sprite);

        // Mostrar el contenido de la ventana
        window.display();

        // Llamar a esta función nuevamente después de un pequeño retraso (aproximadamente 200 ms)
        while (clock.getElapsedTime().asMilliseconds() < 200) {
            // Espera hasta que pase el tiempo deseado
        }
        clock.restart();
    }

    return 0;
}

